orttraining/orttraining/core/graph/optimizer/lamb_optimizer_builder.cc

```diff
@@ -231,11 +280,11 @@ Status LambOptimizerBuilder::Build(
       // w_mixed_precision & w_mixed_precision_new
       if (opt_configs[i].update_weight && opt_configs[i].mixed_precision_weight_arg != nullptr) {
         input_argdefs.emplace_back(ArgDef(
-          opt_configs[i].mixed_precision_weight_arg->Name(),
-          opt_configs[i].mixed_precision_weight_arg->TypeAsProto()));
+            opt_configs[i].mixed_precision_weight_arg->Name(),
```

Comment:
If we are using emplace back, may we not construct the ```ArgDef``` object before passing it. We could leverage in place construction using
```
input_argdefs.emplace_back(opt_configs[i].mixed_precision_weight_arg->Name(), opt_configs[i].mixed_precision_weight_arg->TypeAsProto());
```
Maybe the compiler is smart enough to elide the copy. But if we are using the ```emplace_back```, we could do this.

If ```ArgDef``` is not a heavy object, I think it should be ok. So this is optional.

Reply:
keeping as before